templates/terraform/decoders/kms_crypto_key.go.erb

```diff
@@ -0,0 +1,17 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2018 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+	// Take the returned long form of the name and use it as `self_link`.
+	d.Set("self_link", res["name"].(string))
```

Comment:
Is it possible to reuse the decoder from keyring here?

Reply:
Seems harmless, `res["name"] = d.Get("name").(string)` will have no effect since we don't do anything with the returned name for CryptoKey, though. I'll reuse it.

Reply:
Done.